api_etl/tests/test_individual_import_sink.py

```diff
@@ -54,31 +67,70 @@ def test_init_workflow_service_failure(self, mock_get_workflows):
             'details': 'Service unavailable'
         }
         with self.assertRaises(IndividualImportSink.Error) as context:
-            IndividualImportSink(self.mock_user)
+            IndividualImportSink(self.user)
         self.assertIn('Error occurred: Service unavailable', str(context.exception))
 
     @patch('api_etl.sinks.individual_import_sink.IndividualImportService.import_individuals')
     @patch('api_etl.sinks.individual_import_sink.WorkflowService.get_workflows')
-    def test_push_data(self, mock_get_workflows, mock_import_individuals):
-        mock_get_workflows.return_value = {
-            'success': True,
-            'data': {
-                'workflows': [{'id': 1, 'name': WORKFLOW_NAME}]
-            }
-        }
-        mock_import_individuals.return_value = {'imported': 5}
+    def test_push_data_with_existing_and_new_records(self, mock_get_workflows, mock_import_individuals):
+        mock_get_workflows.side_effect = mock_get_workflow
+        mock_import_individuals.return_value = {'imported': 2}
+
+        sink = IndividualImportSink(self.user)
+        data = [
+            {'external_id': '123', 'name': 'John Doe', 'age': 30},  # Existing record
+            {'external_id': '456', 'name': 'Jane Smith', 'age': 25}  # New record
+        ]
+
+        sink.push(data)
 
-        sink = IndividualImportSink(self.mock_user)
-        data = [{'name': 'John Doe', 'age': 30}, {'name': 'Jane Smith', 'age': 25}]
+        # Check if import_individuals was called twice: once for new, once for existing
+        self.assertEqual(mock_import_individuals.call_count, 2)
```

Comment:
Could you check if there is disable for mssql tests? Here there is currently error in this assertion (mssql context).

Reply:
Fixed with [d811d53](https://github.com/openimis/openimis-be-api_etl_py/pull/7/commits/d811d53dba482994f4d270980dfd5301d38e43e0). In this case, mssql actually caught a worthy problem!